packages/frint-router-react/src/Link.js

```diff
@@ -0,0 +1,118 @@
+import React from 'react'; // eslint-disable-line import/no-extraneous-dependencies
+import PropTypes from 'prop-types'; // eslint-disable-line import/no-extraneous-dependencies
+
+export default class Link extends React.Component {
+  static contextTypes = {
+    app: PropTypes.object.isRequired
+  };
+
+  static propTypes = {
+    to: PropTypes.string.isRequired,
+    exact: PropTypes.bool,
+    type: PropTypes.string,
+    className: PropTypes.string,
+    activeClassName: PropTypes.string,
+    children: PropTypes.node,
+  };
+
+  constructor(...args) {
+    super(...args);
+
+    this.state = {
+      active: false,
+    };
+
+    this.subscription = null;
+  }
+
+  componentDidMount() {
+    this.considerSubscribingToRouter(this.props);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    this.considerSubscribingToRouter(nextProps);
+  }
+
+  componentWillUnmount() {
+    this.unsubscribeFromRouter();
+  }
+
+  considerSubscribingToRouter(nextProps) {
+    if (typeof nextProps.activeClassName === 'string') {
+      if (!this.subscription ||
+          this.props.to !== nextProps.to ||
+          this.props.exact !== nextProps.exact) {
+        this.resubscribeToRouter(nextProps.to, nextProps.exact);
+      }
+    }
+  }
+
+  resubscribeToRouter(to, exact) {
+    this.unsubscribeFromRouter();
+
+    this.subscription = this.context.app
+      .get('router')
+      .getMatch$(to, { exact })
+      .subscribe((matched) => {
+        if (!matched) {
+          return this.setState({ active: false });
+        }
+
+        return this.setState({ active: true });
+      });
+  }
+
+  unsubscribeFromRouter() {
+    if (this.subscription) {
+      this.subscription.unsubscribe();
+    }
+  }
+
+  handleClick = (e) => {
+    e.preventDefault();
+
+    // TODO avoid pushing route if it's current URL
```

Comment:
I've added this TODO, it's the `history` library that issues a warning whenever the same route is pushed. there's a bit of discussion in here about whether it should warn or stay silent: https://github.com/ReactTraining/react-router/issues/4467. but I guess `<Link>` shouldn't push the url if it's `path` exactly matches current url